### PR TITLE
[NZPMC-006] Created database schema / basic graphql files

### PR DESCRIPTION
**Pull request by Kingston802**
**Issue**

The previous schema was an example one.

**Solution**

Implemented the schema described in CZ's design document: [link](https://docs.google.com/document/d/19tFyLQHdJR7-pJrncysTQysQaJuDGlJADo9HaH7ZQFo/edit#
)

**Risk**

Low risk. 

**Reviewers**

Scorpionknifes
Will

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,11 +2,14 @@ import { ApolloServer } from 'apollo-server'
 import { readFileSync } from 'fs'
 import resolvers from './resolvers'
 
-const typeDefs = readFileSync('./schemas/schema.graphql').toString('utf-8')
+const queries = readFileSync('./schemas/schema.graphql').toString('utf-8')
+// const mutations = readFileSync('./schemas/mutations.graphql').toString('utf-8')
+const objects = readFileSync('./schemas/objects.graphql').toString('utf-8')
+// const inputs = readFileSync('./schemas/inputs.graphql').toString('utf-8')
```
Comment (scorpionknifes) [inline index.js:8]: Suggestion:
use `fs.readdirSync` with wildcard `*.graphql` to get an array of `filepaths`
use `filepaths.map(()=>...)` to create a typeDefs

File: schemas/mutations.graphql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+type Mutation {
+
```
Comment (scorpionknifes) [inline schemas/mutations.graphql:2]: Implement your first mutation example: 
- `createQuestion(form: CreateQuestion): Question` with `Input CreateQuestion` in `input.graphql` file

File: schemas/objects.graphql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    firstName: String
+    lastName: String
+    yearLevel: String
+    role: String 
+    created: DateTime 
+    modified: DateTime 
+}
+
+type Question {
+    id: ID!
```
Comment (scorpionknifes) [inline schemas/objects.graphql:16]: Let's define all the variables from the database schema.

also, define a nested query e.g

```gql
type Quiz {
	...
	questions: [Question]
	...
}
```

some of the nested query you can implement:
- Quiz -> Question
- Question -> Options
- UserQuiz -> Quiz, User (remember to remove UserAnswer)
- UserQuiz -> UserQuizAnswer

Comment (scorpionknifes) [reply schemas/objects.graphql:16]: We can also do queries inside types e.g
```gql
type Quiz {
	...
	questions: [Question]
	"Get question by index"
	question(index: Int!): Question 
	...
}
```

File: schemas/schema.graphql [hunk 4]
```diff
@@ -1,4 +1,12 @@
...
-    numberSix: Int!
-    numberSeven: Int!
+    me: User
+    user(userID: ID!): User 
+    users: [User]
+    question(questionID: ID!): Question
+    questions: [Question]
+    quizzes: [Quiz]
+    userQuiz: UserQuiz
+    userQuizAnswers(quizID: ID!): [UserQuizAnswer]
```
Comment (scorpionknifes) [inline schemas/schema.graphql:11]: Let's add another query - `userQuizzes: [UserQuiz]`
And change - `userQuiz(quizID: ID!): UserQuiz`
Remove `userQuizAnswers`

This is cause we are adding UserQuizAnswer inside

**Review by scorpionknifes — CHANGES_REQUESTED** [review]
Great Work!

Just a couple of things to do more for this PR!

**Comment by Kingston802** [comment]
Completed most changes but need to discuss nested queries and objects (UserAnswer vs UserQuizAnswer) with CZ.

**Review by scorpionknifes — APPROVED** [review]
LGTM

**Review by Will-ZYS — APPROVED** [review]
LGTM

**Pull request merged**